contracts/StableJoeStaking.sol

```diff
@@ -0,0 +1,312 @@
+        }
+
+        internalJoeBalance = internalJoeBalance.add(amountMinusFee);
+        joe.safeTransferFrom(msg.sender, feeCollector, fee);
+        joe.safeTransferFrom(msg.sender, address(this), amountMinusFee);
+        emit Deposit(msg.sender, amountMinusFee);
+    }
+
+    /**
+     * @notice get user info
+     * @param _user The address of the user
+     * @param _rewardToken The address of the reward token
+     * @return uint256 the user's amount
+     * @return uint256 the user's reward debt of the reward token chosen
+     */
+    function getUserInfo(address _user, IERC20Upgradeable _rewardToken) external view returns (uint256, uint256) {
+        UserInfo storage user = userInfo[_user];
+        return (user.amount, user.rewardDebt[_rewardToken]);
+    }
+
+    /**
+     * @notice get the number of reward tokens
+     * @return The length of the array
+     */
+    function rewardTokensLength() external view returns (uint256) {
+        return rewardTokens.length;
+    }
+
+    /**
+     * @notice Set the reward token
+     * @param _rewardToken The address of the reward token
+     */
+    function addRewardToken(IERC20Upgradeable _rewardToken) external onlyOwner {
+        require(
+            !isRewardToken[_rewardToken] && address(_rewardToken) != address(0),
+            "StableJoeStaking: token can't be added"
+        );
+        rewardTokens.push(_rewardToken);
+        isRewardToken[_rewardToken] = true;
+        updateReward(_rewardToken);
+    }
+
+    /**
+     * @notice Remove a reward token
+     * @param _rewardToken The address of the reward token
+     */
+    function removeRewardToken(IERC20Upgradeable _rewardToken) external onlyOwner {
+        require(isRewardToken[_rewardToken], "StableJoeStaking: token can't be removed");
+        updateReward(_rewardToken);
+        isRewardToken[_rewardToken] = false;
+        uint256 len = rewardTokens.length;
+        for (uint256 i; i < len; i++) {
+            if (rewardTokens[i] == _rewardToken) {
+                rewardTokens[i] = rewardTokens[len - 1];
+                rewardTokens.pop();
+                break;
+            }
+        }
+    }
+
+    /**
+     * @notice Set the deposit fee percent
+     * @param _depositFeePercent The new deposit fee percent
+     */
+    function setdepositFeePercent(uint256 _depositFeePercent) external onlyOwner {
+        require(_depositFeePercent < 5e17, "StableJoeStaking: deposit fee can't be greater than 50%");
+        depositFeePercent = _depositFeePercent;
+    }
+
+    /**
+     * @notice View function to see pending reward token on frontend
+     * @param _user The address of the user
+     * @return `_user`'s pending reward token
+     */
+    function pendingReward(address _user, IERC20Upgradeable _token) external view returns (uint256) {
+        require(isRewardToken[_token], "StableJoeStaking: wrong reward token");
+        UserInfo storage user = userInfo[_user];
+        uint256 totalJoe = internalJoeBalance;
```

Comment:
any reason for the variable `totalJoe` over just using `internalJoeBalance` inline. I've also seen a pattern where non state variables are prefixed with `_`. I like this as you immediately know if a variable is a memory or storage variable.

Reply:
I allign on this